### PR TITLE
feat: enable windows control overlay on Windows

### PR DESCRIPTION
**Pull request by mlaurencin**
Backport of #29600.

See that PR for details.

Notes: Added support for Windows Control Overlay on Windows.

### CODE REVIEW

**Review by MarshallOfSound — COMMENTED** [review]
Approved by @electron/wg-releases, but I want to review the actual patch files before approve-approving

**Review by zcbenz — APPROVED** [review]

File: shell/browser/native_window_mac.mm
Comment (zcbenz) [inline shell/browser/native_window_mac.mm:255]: This code has no effect as it is reading the option into a temporary variable.

**Review by zcbenz — CHANGES_REQUESTED** [review]

**Review by zcbenz — APPROVED** [review]

**Comment by zcbenz** [comment]
This PR requires review from @electron/wg-upgrades.

**Review by deepak1556 — APPROVED** [review]
@mlaurencin are the chromium patches direct backports or did you have to make any modifications ?

**Comment by mlaurencin** [comment]
> @mlaurencin are the chromium patches direct backports or did you have to make any modifications ?

The chromium patches were just directly backported with no modifications.

**Pull request merged**
